src/Middleware/RequestThrottling/test/MiddlewareTests.cs

```diff
@@ -146,5 +146,19 @@ public async void FullQueueInvokesOnRejected()
             await middleware.Invoke(context).OrTimeout();
             Assert.True(onRejectedInvoked);
         }
+
+        [Fact]
+        public async void ExceptionThrownDuringOnRejected()
```

Comment:
@halter73 @DylanDmitri I'm not sure if I quite understand the assertion part of the test that is needed. Is this what you had in mind?

Reply:
looks good!